vpr/src/device/rr_graph_obj.h

```diff
@@ -0,0 +1,647 @@
+
+    /* A generic function to set the ptc_num for a node */
+    void set_node_ptc_num(const RRNodeId& node, const short& ptc);
+    /* Only applicable to IPIN and OPIN, set the ptc_num for a node, which is the pin id in a logic block,
+     * See definition in t_type_descriptor data structure   
+     */
+    void set_node_pin_num(const RRNodeId& node, const short& pin_id);
+    /* Only applicable to CHANX and CHANY, set the ptc_num for a node, which is the track id in a routing channel,
+     * Routing channel is a group of routing tracks, each of which has a unique index
+     * An example 
+     *       Routing Channel 
+     *    +------------------+
+     *    |                  |
+     * ---|----------------->|----> track_id: 0
+     *    |                  |
+     *   ...   More tracks  ...
+     *    |                  |
+     * ---|----------------->|----> track_id: 1
+     *    |                  |
+     *    +------------------+
+     */
+    void set_node_track_num(const RRNodeId& node, const short& track_id);
+    /* Only applicable to SOURCE and SINK, set the ptc_num for a node, which is the class number in a logic block,
+     * See definition in t_type_descriptor data structure   
+     */
+    void set_node_class_num(const RRNodeId& node, const short& class_id);
+
+    /* Set the routing cost index for node, see node_cost_index() for details */
+    /* TODO the cost index should be changed to a StrongId!!! */
+    void set_node_cost_index(const RRNodeId& node, const short& cost_index);
+    /* Set the directionality for a node, only applicable to CHANX and CHANY */
+    void set_node_direction(const RRNodeId& node, const e_direction& direction);
+    /* Set the side for a node, only applicable to OPIN and IPIN */
+    void set_node_side(const RRNodeId& node, const e_side& side);
+    /* Set the RC information for a node */
+    void set_node_R(const RRNodeId& node, const float& R);
+    void set_node_C(const RRNodeId& node, const float& C);
+    /* Set the routing segment linked to a node, only applicable to CHANX and CHANY */
+    void set_node_segment(const RRNodeId& node, const RRSegmentId& segment_index);
+
+    /* Edge related */
+    /* classify the input edges of each node to be configurable (1st part) and non-configurable (2nd part) */
+    void partition_node_in_edges(const RRNodeId& node);
+    /* classify the output edges of each node to be configurable (1st part) and non-configurable (2nd part) */
+    void partition_node_out_edges(const RRNodeId& node);
+    void partition_in_edges();  /* classify the input edges of each node to be configurable (1st part) and non-configurable (2nd part) */
+    void partition_out_edges(); /* classify the output edges of each node to be configurable (1st part) and non-configurable (2nd part) */
+    void partition_edges();     /* classify the edges of each node to be configurable (1st part) and non-configurable (2nd part) */
```

Comment:
Suggest only exposing the partition_edges() function in the public interface.  Not clear that any external code should ever call the lower level functions, so I think they should be private.